actions/gha-workflow-validator/src/index.ts

```diff
@@ -0,0 +1,72 @@
+import * as core from "@actions/core";
+import * as github from "@actions/github";
+import { commentOnPrOrUpdateExisting, deleteCommentOnPRIfExists, getComparison } from "./github.js";
+import { validateActionReferenceChanges } from "./action-reference-validations.js";
+import { filterForGithubWorkflowChanges, formatGithubComment, parseAllAdditions } from "./utils.js";
+import { COMMENT_HEADER } from "./strings.js";
+
+(async () => {
+  const { token, owner, repo, base, head, prNumber } = getInvokeContext();
+  const octokit = github.getOctokit(token);
+
+  const allFiles = await getComparison(octokit, owner, repo, base, head);
+  const ghaWorkflowFiles = filterForGithubWorkflowChanges(allFiles);
```

Comment:
Same as above, we could use the aforementioned GHA to delegate this logic

Reply:
I don't see the benefit of using an external action to handle the computation of changed files, and using that as input.

The goal of this action is to validate changes to workflow files and only getting a list of modified files kind of defeats that right? Because you still don't know what was changed in that file directly.

For that use-case, the GH API compare endpoint is better for this use-case isn't it? It returns all files modified and the patch for each file.

Unless you're suggesting the linting happens for the whole file?

Reply:
Good point, I didnt realize you're doing a line diff rather than a file diff